src/transactions/instances/simpleTransaction.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+/* eslint-disable @typescript-eslint/naming-convention */
+
+import { Deserializer } from "../../bcs/deserializer";
+import { Serializable, Serializer } from "../../bcs/serializer";
+import { AccountAddress } from "../../core";
+import { RawTransaction } from "./rawTransaction";
+
+/**
+ * Representation of a SimpleTransaction that can serialized and deserialized
+ */
+export class SimpleTransaction extends Serializable {
+  public rawTransaction: RawTransaction;
+
+  public feePayerAddress?: AccountAddress | undefined;
+
+  // We dont really need it, we add it for type checkings we do
+  // throughout the SDK
+  public readonly secondarySignerAddresses: undefined;
+
+  /**
+   * SimpleTransaction represents a simple transaction type of a single signer that
+   * can be submitted to Aptos chain for execution.
+   *
+   * SimpleTransaction metadata contains the Raw Transaction and an optional
+   * sponsor Account Address to pay the gas fees.
+   *
+   * @param rawTransaction The Raw Tranasaction
+   * @param feePayerAddress The sponsor Account Address
+   */
+  constructor(rawTransaction: RawTransaction, feePayerAddress?: AccountAddress) {
+    super();
+    this.rawTransaction = rawTransaction;
+    this.feePayerAddress = feePayerAddress;
+  }
+
+  serialize(serializer: Serializer): void {
+    this.rawTransaction.serialize(serializer);
+
+    if (this.feePayerAddress === undefined) {
+      serializer.serializeBool(false);
+    } else {
+      serializer.serializeBool(true);
+      this.feePayerAddress.serialize(serializer);
+    }
```

Comment:
Interesting, using a bool instead of `null` or `[]`

Though... actually I think that 

`bool,address` serializes the same in BCS by accident as the "optional" address
`[address]` and `[]`

Up to you here, because you could just serialize a vector of 0 for none, and a vector of 1 for some.

Reply:
yea I think you are right, we use `bool` [here](https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/transactions/instances/transactionPayload.ts#L356) so thought just going with it